src/core/unit.c

```diff
@@ -3612,37 +3612,78 @@ static int get_name_owner_handler(sd_bus_message *message, void *userdata, sd_bu
 }
 
 int unit_install_bus_match(Unit *u, sd_bus *bus, const char *name) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m1 = NULL, *m2 = NULL;
+        Service *s = SERVICE(u);
         const char *match;
+        usec_t timeout_usec;
         int r;
 
         assert(u);
         assert(bus);
         assert(name);
+        assert(s);
 
         if (u->match_bus_slot || u->get_name_owner_slot)
                 return -EBUSY;
 
+        /* NameOwnerChanged and GetNameOwner is used to detect when
+         * a service finished starting up. The dbus call timeout
+         * shouldn't be earlier than that. */
```

Comment:
btw, we generally line break at 109 characters, see .editorconfig and the relevant files for vim + emacs